src/backend.rs

```diff
@@ -327,8 +341,40 @@ pub fn copy_incoming_arg(ctx: &mut Context, arg_pos: u32) {
     // And then move a value from a register into local variable area on the stack.
     let offset = sp_relative_offset(ctx, arg_pos);
     dynasm!(ctx.asm
-        ; mov [rsp + offset], Rq(reg) 
+        ; mov [rsp + offset], Rq(reg)
+    );
+}
+
+pub fn pass_outgoing_args(ctx: &mut Context, arity: u32) {
+    for arg_pos in (0..arity).rev() {
+        ctx.sp_depth.free(1);
+
+        let loc = abi_loc_for_arg(arg_pos);
+        match loc {
+            ArgLocation::Reg(gpr) => {
+                dynasm!(ctx.asm
+                    ; pop Rq(gpr)
+                );
+            }
+            _ => unimplemented!("don't know how to pass argument {} via {:?}", arg_pos, loc),
+        }
+    }
+}
+
+pub fn call_direct(ctx: &mut Context, index: u32, return_arity: u32) {
+    assert!(return_arity == 0 || return_arity == 1);
+
+    let label = &ctx.func_starts[index as usize].1;
+    dynasm!(ctx.asm
+        ; call =>*label
     );
+
+    if return_arity == 1 {
```

Comment:
Could you add an `assert!(return_arity <= 1)` for now, so that we don't silently misbehave if someone starts working on multi-value support?

Reply:
do you mean another one? : )

Reply:
Oops, you're right :-).